### PR TITLE
Fixed incorrect type formats in SQL database

### PR DESCRIPTION
**Pull request by jaredmoo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sameralameer** [comment]
Apologies for the Delay @jaredmoo, we lately had a surge of PRs, someone will get on this shortly

**Comment by jaredmoo** [comment]
Thanks Samer :)

**Comment by salameer** [comment]
Gotcha

**Comment by balajikris** [comment]
I ran the linter and semantic check tools and this change appears to be clean. I found no errors from the tools. Will sign off, after I do a manual review.

**Review by balajikris — DISMISSED** [review]
LGTM.

**Comment by balajikris** [comment]
okay my bad here. I forgot to pass in the composite swagger entry point to the tool and the tool happily returned no errors. If I do however pass in the compositeSwagger entry point, it throws a bunch of errors. 

I'll go through the list again. 👍

**Comment by balajikris** [comment]
**Status as of 4/13:** emailed @jaredmoo about the list of linter errors and examples errors that we should consider addressing.

File: arm-sql/2014-04-01/swagger/capabilities.json [hunk 1]
```diff
@@ -118,7 +119,9 @@
                 "id": {
                     "readOnly": true,
                     "type": "string",
-                    "description": "The unique ID of the service objective."
+                    "format": "uuid",
```
Comment (balajikris) [inline arm-sql/2014-04-01/swagger/capabilities.json]: autorest linter tags this as

`WARNING: Guid used at the #/Definitions/ServiceObjectiveCapability/.../id. Usage of Guid is not recommanded. If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.`

@ravbhatnagar : what's the recommended change here? I do not have enough context on why guids are bad and what's the right way to fix this. I'll appreciate it if you can fill me and Jared in here.

**Review by balajikris — CHANGES_REQUESTED** [review]
Emailed a set of linter errors and example errors that we need to address.

File: arm-sql/2014-04-01/swagger/capabilities.json (hunk 1 above)
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/capabilities.json]: Unfortunately no change is possible. This is what the service is returning, and changing it at this point would be a REST API breaking change. We will have to consider removing this guid from the next api version.

Comment (balajikris) [inline arm-sql/2014-04-01/swagger/capabilities.json]: Hmm. Thought as much. Thanks for confirming

**Comment by balajikris** [comment]
**Summary of errors we will track by bugs:**

1.  _ERROR: Please provide x-ms-examples describing minimum/maximum property set for response/request payloads for operations.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/paths_
**[Jared Moore] Let’s file a separate issue for this**

1.  _ERROR: The tracked resource, 'Database', must have a list by resource group operation.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/Database
ERROR: The tracked resource, 'ElasticPool', must have a list by resource group operation.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/ElasticPool
ERROR: The tracked resource, 'Database', must have a list by subscriptions operation.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/Database_
**[Jared Moore] Let’s file a separate issue for this. It’s not implemented in the service right now.**

1.  _ERROR: The tracked resource, 'ElasticPool', must have a list by subscriptions operation.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/ElasticPool
ERROR: Tracked resource 'Database' must have patch operation that at least supports the update of tags.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/Database
ERROR: Tracked resource 'Server' must have patch operation that at least supports the update of tags.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/Server
ERROR: Tracked resource 'ElasticPool' must have patch operation that at least supports the update of tags.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/ElasticPool_
**[Jared Moore] Let’s file a separate issue for this. It’s not implemented in the service right now.**

1. _ERROR: Model definition 'FirewallRule' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
        Path: file:///C:/workspace/azure-rest-api-specs/arm-sql/compositeSql.json#/definitions/FirewallRule_
**[Jared Moore] The service really doesn’t send ‘type’ for FirewallRule. This is a bug but not one that I can easily fix. Will file issue for this.**

**Review by balajikris — APPROVED** [review]

**Comment by jaredmoo** [comment]
Thanks, let me just double check there are no more changes needed

**Comment by balajikris** [comment]
Reviewed again after the last commit. things are still good 👍

**Pull request merged**